### PR TITLE
mysql-connector-odbc 5.3.8 (new formula)

### PR DESCRIPTION
**Pull request by akshayganeshen**
Added CMake-based build process for MySQL ODBC connector. So far it
only supports static linking against MySQL client & OpenSSL libraries.

This formula is meant for unixodbc users only. The pre-built drivers
available from MySQL are meant to work with libiodbc, so users can
just download those if they work with iODBC.

### CODE REVIEW

File: Formula/mysql-connector-odbc.rb [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+
+    # For static linking:
+    ssl_libs = "#{Formula["openssl"].opt_lib}/libssl.a "
+    ssl_libs << "#{Formula["openssl"].opt_lib}/libcrypto.a"
+    args << "-DMYSQL_EXTRA_LIBRARIES=#{ssl_libs}"
+
+    system "cmake", ".", *args
+    # There are parallel build issues for the tests, and there is no easy way
+    # to disable to tests. Have to use a workaround...
+    # Use separate invocations of make to build in the right order:
```
Comment (bfontaine) [inline Formula/mysql-connector-odbc.rb]: Please report these issues to upstream.

**Review by bfontaine — CHANGES_REQUESTED** [review]

File: Formula/mysql-connector-odbc.rb [hunk 2]
```diff
@@ -0,0 +1,42 @@
...
+    #        the MYSQL_INCLUDE_DIR and MYSQL_LIB_DIR settings.
+    #        This can lead to errors during the link step, so avoid it:
+    # args << "-DMYSQL_DIR=#{Formula["mysql"].opt_prefix}"
+    args << "-DMYSQL_INCLUDE_DIR=#{Formula["mysql"].opt_include}/mysql"
+    args << "-DMYSQL_LIB_DIR=#{Formula["mysql"].opt_lib}"
+    args << "-DMYSQLCLIENT_STATIC_LINKING=1"
+    args << "-DMYSQL_LINK_FLAGS=-L#{Formula["mysql"].opt_lib}"
+
+    # For static linking:
+    ssl_libs = "#{Formula["openssl"].opt_lib}/libssl.a "
```
Comment (DomT4) [inline Formula/mysql-connector-odbc.rb]: Static linking of SSL libraries is a grim grim thing.

**Comment by OmgLag** [comment]
Thanks for the feedback!

@bfontaine I'll report the issue, but for the time being, I think it's acceptable to build it this way. Using no-parallel would also work, but this solution still allows building it in parallel. If that's unacceptable, I will wait until it is resolved upstream and then update to the newer version. Please let me know!

@DomT4 Very true. I've switched it over to using the shared libraries for mysqlclient and openssl. Hopefully it's a little less grim...

**Comment by bfontaine** [comment]
Upstream should at least be aware of the issue.

**Review by ilovezfs — CHANGES_REQUESTED** [review]
Please report the parallelization issue upstream, as that's a bug that will block inclusion of this formula in Homebrew core. Another option is not to run the tests, but that will also apparently require a change upstream since you said there's no way to disable the tests currently.

**Pull request closed without merging**